circuit/src/contracts-registry/src/tests.rs

```diff
@@ -336,3 +337,238 @@ fn fetch_contracts_with_no_parameters_should_error() {
+    let origin = Origin::signed(1);
+    let test_contract = RegistryContract {
+        code_txt: b"some_code".to_vec(),
+        bytes: vec![],
+        author: 1_u64,
+        author_fees_per_single_use: None,
+        abi: None,
+        action_descriptions: vec![],
+        info: None,
+        meta: ContractMetadata::new(
+            vec![],
+            b"contract 1".to_vec(),
+            vec![],
+            vec![],
+            None,
+            None,
+            None,
+            None,
+            None,
+        ),
+    };
+
+    let requester = 1_u64;
+
+    ExtBuilder::default()
+        .with_contracts(vec![test_contract.clone()])
+        .build()
+        .execute_with(|| {
+            assert_err!(
+                ContractsRegistry::add_new_contract(origin, requester, test_contract.clone()),
+                DispatchError::BadOrigin
+            );
+        })
+}
+
+#[test]
+fn add_new_contract_fails_if_contract_already_exists() {
+    let origin = Origin::root();
+    let test_contract = RegistryContract {
+        code_txt: b"some_code".to_vec(),
+        bytes: vec![],
+        author: 1_u64,
+        author_fees_per_single_use: None,
+        abi: None,
+        action_descriptions: vec![],
+        info: None,
+        meta: ContractMetadata::new(
+            vec![],
+            b"contract 1".to_vec(),
+            vec![],
+            vec![],
+            None,
+            None,
+            None,
+            None,
+            None,
+        ),
+    };
+
+    let requester = 1_u64;
+
+    ExtBuilder::default().build().execute_with(|| {
+        crate::ContractsRegistry::<Test>::insert(
+            test_contract.generate_id::<Test>(),
+            test_contract.clone(),
+        );
+        assert_err!(
+            ContractsRegistry::add_new_contract(origin, requester, test_contract.clone()),
+            Error::<Test>::ContractAlreadyExists
+        )
+    })
+}
+
+#[test]
+fn purge_success() {
+    let origin = Origin::root();
+    let requester = 1_u64;
+
+    let test_contract = RegistryContract {
+        code_txt: b"some_code".to_vec(),
+        bytes: vec![],
+        author: 1_u64,
+        author_fees_per_single_use: None,
+        abi: None,
+        action_descriptions: vec![],
+        info: None,
+        meta: ContractMetadata::new(
+            vec![],
+            b"contract 1".to_vec(),
+            vec![],
+            vec![],
+            None,
+            None,
+            None,
+            None,
+            None,
+        ),
+    };
+
+    let contract_id = test_contract.generate_id::<Test>();
+
+    ExtBuilder::default()
+        .with_contracts(vec![test_contract.clone()])
+        .build()
+        .execute_with(|| {
+            crate::ContractsRegistry::<Test>::insert(
+                test_contract.generate_id::<Test>(),
+                test_contract.clone(),
+            );
+            assert_ok!(ContractsRegistry::purge(origin, requester, contract_id));
```

Comment:
Check here that `ContractsRegistry::get(contract_id)` returns none

Reply:
Fixed on the latest commit